### DISCUSSION TITLE
Poetry explicit source with private artifactory and sub-dependencies

### DISCUSSION THREAD

**Discussion by dmoshayof** [python-poetry/poetry#8224] [Q&A]
Hi,
I'm trying to use the latest option for source configuration and use "explicit" for my private artifactory URL.
The "Dependencies resolving stage" is very fast with this config and I wish to use it always, but I doesn't work for me when I have private-sub-dependencies.
For Example:

[tool.poetry.dependencies]
python = "^3.9"
data-tools = {version = "1.1.28", source = "private-repo"}

[[tool.poetry.source]]
name = "private-repo"
url = "<URL>/artifactory/api/pypi/pypi-dev/simple/"
priority= "explicit"

in "data-tools" there are some others package from "private-repo" which poetry can't find when I'm using "EXPLICIT".
I tried also "supplemental"/"secondary" and it takes more than 2000sec to resolve.....which is not a solution for me.
When I added "data-tools" private sub-dependencies to the same "toml" with explicit, it worked well, but I don't think it's a right approach. 

any ideas?

**Comment by radoering** [comment]
> When I added "data-tools" private sub-dependencies to the same "toml" with explicit, it worked well, but I don't think it's a right approach.

Apart from using another priority such as "supplemental" that is the right approach, in other words I don't think there is a better one. Personally, I'd put all these transitive dependencies in a separate (optional) group with the version constraint `"*"`.

**Reply by dmoshayof** [reply]
I tried to use "supplemental" and after 4000sec I aborted it. It does not make sense to wait that long.
